app/V1Module/presenters/UploadedFilesPresenter.php

```diff
@@ -99,7 +100,14 @@ public function actionContent(string $id) {
     if (!$this->uploadedFileAcl->canDownload($file)) {
       throw new ForbiddenRequestException("You are not allowed to access file '{$file->getId()}");
     }
-    $this->sendSuccessResponse($file->getContent());
+
+    $content = $file->getContent();
+    $fixedContent = Encoding::toUTF8($content);
```

Comment:
I would suggest to limit maximal size of the returned text (since it is kept in memory). If the file is larger than the limit, only it leading part (up to the limit) will be returned and "incomplete" flag will be added.